### DISCUSSION TITLE
求教：在 3.x中有dynamic() 方法，在4.x中没有了，请问有替换的方案么？

### DISCUSSION THREAD

**Discussion by etechi** [umijs/umi#11016] [Q&A] [answered]
3.x 时可以使用
import { dynamic } from 'umi';
改为
import { dynamic } from '@umijs/max';

后,运行时提示
TypeError: (0 , _umijs_max__WEBPACK_IMPORTED_MODULE_4__.dynamic) is not a function

**Comment by fz6m** [comment, marked as answer]
umi 3 为了兼容 ie 所以 `dynamic` 用的 loadable 支持动态分包加载，umi 4 不考虑 ie ，所以这个 api 无了，直接自己用 react 原生懒加载即可：

```tsx
import { Suspense, lazy } from 'react'

const C = lazy(() => import('./path/to/component'))

// ...

return (
  <Suspense fallback={<div>loading...</div>}>
    <C />
  </Suspense>
)
```

**Reply by etechi** [reply]
非常感谢！
